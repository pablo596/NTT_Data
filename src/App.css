#root {
  display: flex;
  flex: 1;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
  background-color: #f4f6f9;
}

.logo {
  height: 6em;
  padding: 1.5em;
  will-change: filter;
  transition: filter 300ms;
}
.logo:hover {
  filter: drop-shadow(0 0 2em #646cffaa);
}
.logo.react:hover {
  filter: drop-shadow(0 0 2em #61dafbaa);
}

@keyframes logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@media (prefers-reduced-motion: no-preference) {
  a:nth-of-type(2) .logo {
    animation: logo-spin infinite 20s linear;
  }
}

.card {
  padding: 2em;
}

.read-the-docs {
  color: #888;
}

.container {
  margin: auto;
  width: 90%;
  padding: 10px;
}

.input {
  border: 1px solid #c8c8c8;
  border-radius: 7px;
  height: 22.5px;
  width: 200px;
  padding: 10px;
}
.input:focus {
  transition: 0.25s ease-in;
  outline: none !important;
  border: 1px solid #4a4a4a;
}

.button {
  background-color: #f9de4b;
  border-radius: 3px;
}

.secondary-button {
  background-color: #e9ecf3;
}

.columns {
  display: flex;
  flex-direction: row;
}
