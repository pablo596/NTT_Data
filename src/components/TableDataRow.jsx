import { formatDateDisplay } from '../helpers';

import Popup from 'reactjs-popup';
import PropTypes from 'prop-types';

export const TableDataRow = ({
  productoFinanciero,
  eliminarRegistro,
  editarRegistro,
}) => {
  return (
    <tr key={productoFinanciero.id}>
      <td>
        <img
          className="logo_producto_financiero"
          src={productoFinanciero.logo}
          width={100}
        />
      </td>
      <td>{productoFinanciero.name}</td>
      <td>{productoFinanciero.description}</td>
      <td>
        {formatDateDisplay(
          new Date(productoFinanciero.date_release).toUTCString()
        )}
      </td>
      <td>
        {formatDateDisplay(
          new Date(productoFinanciero.date_revision).toUTCString()
        )}
      </td>
      <td>
        <div>
          <Popup
            trigger={
              <button>
                <i className="bx bx-dots-vertical-rounded "></i>
              </button>
            }
            position="bottom center"
          >
            <div className="popup_menu">
              <button onClick={() => editarRegistro()}>Editar</button>
              <button onClick={() => eliminarRegistro(productoFinanciero.id)}>
                Eliminar
              </button>
            </div>
          </Popup>
        </div>
      </td>
    </tr>
  );
};

TableDataRow.propTypes = {
  productoFinanciero: PropTypes.shape({
    id: PropTypes.string,
    logo: PropTypes.string,
    name: PropTypes.string,
    description: PropTypes.string,
    date_release: PropTypes.string,
    date_revision: PropTypes.string,
  }),
  eliminarRegistro: PropTypes.func,
  editarRegistro: PropTypes.func,
};
