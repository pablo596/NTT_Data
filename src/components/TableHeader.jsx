export const TableHeader = () => {
  return (
    <thead>
      <tr>
        <th>Logo</th>
        <th>Nombre</th>
        <th>
          Descripción <i className="bx bxs-info-circle bx-flip-horizontal"></i>
        </th>
        <th>
          Fecha de liberación{' '}
          <i className="bx bxs-info-circle bx-flip-horizontal"></i>
        </th>
        <th>
          Fecha de reestructuración{' '}
          <i className="bx bxs-info-circle bx-flip-horizontal"></i>
        </th>
        <th></th>
      </tr>
    </thead>
  );
};
