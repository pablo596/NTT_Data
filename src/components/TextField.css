.textfield {
  display: flex;
  flex-direction: column;
  justify-content: left;
  align-items: flex-start;
  width: 100%;
  margin-bottom: 5%;
}

.label {
  font-size: 0.9rem;
}

.input-form {
  width: 85%;
}

.error-message {
  color: #c32b1f;
  margin-top: 0.5rem;
  font-size: 0.7rem;
}

.input-form.error-message,
.input-form.error-message:focus {
  border: 1px solid #c32b1f;
}
