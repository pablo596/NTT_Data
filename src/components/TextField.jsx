import PropTypes from 'prop-types';

export const TextField = ({
  label,
  placeholder,
  type,
  name,
  error,
  helperText,
  disabled,
  value,
  minimun,
  onChange,
}) => {
  return (
    <div className="textfield">
      <label htmlFor={name} className="label">
        {label}
      </label>
      <input
        id={name}
        type={type}
        name={name}
        onChange={onChange}
        className={`input input-form ${error ? 'error-message' : ''}`}
        placeholder={placeholder}
        disabled={disabled}
        value={value}
        min={minimun}
        lang="en-GB"
      />
      {error ? (
        <span className={`${error ? 'error-message' : ''}`}>{helperText}</span>
      ) : null}
    </div>
  );
};

TextField.propTypes = {
  label: PropTypes.string,
  placeholder: PropTypes.string,
  type: PropTypes.string,
  name: PropTypes.string,
  error: PropTypes.bool,
  helperText: PropTypes.string,
  disabled: PropTypes.bool,
  value: PropTypes.string,
  minimun: PropTypes.string,
  onChange: PropTypes.func,
};
