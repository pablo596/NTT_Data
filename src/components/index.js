export * from './TableHeader';
export { TextField } from './TextField';
export * from './TableDataRow';
