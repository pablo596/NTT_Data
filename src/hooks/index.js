export * from './useFetchProductosFinancieros';
export * from './useDebounce';
export * from './useForm';
