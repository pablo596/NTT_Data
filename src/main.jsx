import React from 'react';
import ReactDOM from 'react-dom/client';
import { BrowserRouter } from 'react-router-dom';

import './index.css';
import './App.css';
import './components/TextField.css';
import './views/TablaProductosFinancieros.css';
import './views/FormularioProductoFinanciero.css';
import { ProductosFinancierosApp } from './ProductosFinancierosApp';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <BrowserRouter>
      <ProductosFinancierosApp />
    </BrowserRouter>
  </React.StrictMode>
);
