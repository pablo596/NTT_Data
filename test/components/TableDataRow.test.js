import { render, screen } from '@testing-library/react';
import { TableDataRow } from '../../src/components';

describe('Pruebas en <TableDataRow/>', () => {
  const productoFinanciero = {
    id: 'trj-crd',
    logo: 'https://www.visa.com.ec/dam/VCOM/regional/lac/SPA/Default/Pay%20With%20Visa/Tarjetas/visa-signature-400x225.jpg',
    name: 'Tarjetas de Crédito',
    description: 'Tarjeta de consumo bajo la modalidad de crédito',
    date_release: '2023-02-01',
    date_revision: '2024-02-01',
  };
  const eliminarRegistro = () => {};
  test('Debe de hacer match con el snapshot', () => {
    const { container } = render(
      <TableDataRow
        productoFinanciero={productoFinanciero}
        eliminarRegistro={eliminarRegistro}
      />
    );

    expect(container).toMatchSnapshot();
  });

  test('Debe de mostrar la imagen con el URL', () => {
    render(
      <TableDataRow
        productoFinanciero={productoFinanciero}
        eliminarRegistro={eliminarRegistro}
      />
    );

    expect(screen.getByRole('img').src).toBe(productoFinanciero.logo);
  });
});
