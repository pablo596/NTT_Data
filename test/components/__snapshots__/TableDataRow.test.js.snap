// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TableDataRow/> Debe de hacer match con el snapshot 1`] = `
<div>
  <tr>
    <td>
      <img
        class="logo_producto_financiero"
        src="https://www.visa.com.ec/dam/VCOM/regional/lac/SPA/Default/Pay%20With%20Visa/Tarjetas/visa-signature-400x225.jpg"
        width="100"
      />
    </td>
    <td>
      Tarjetas de Crédito
    </td>
    <td>
      Tarjeta de consumo bajo la modalidad de crédito
    </td>
    <td>
      01/02/2023
    </td>
    <td>
      01/02/2024
    </td>
    <td>
      <div>
        <button
          aria-describedby="popup-1"
        >
          <i
            class="bx bx-dots-vertical-rounded "
          />
        </button>
      </div>
    </td>
  </tr>
</div>
`;
